testing/util/lightclient.go

```diff
@@ -66,37 +66,71 @@ func (l *TestLightClient) SetupTestCapella() *TestLightClient {
 	parentRoot, err := signedParent.Block().HashTreeRoot()
 	require.NoError(l.T, err)
 
-	block := NewBeaconBlockCapella()
-	block.Block.Slot = slot
-	block.Block.ParentRoot = parentRoot[:]
+	if blinded {
+		block := NewBlindedBeaconBlockCapella()
+		block.Block.Slot = slot
+		block.Block.ParentRoot = parentRoot[:]
 
-	for i := uint64(0); i < params.BeaconConfig().MinSyncCommitteeParticipants; i++ {
-		block.Block.Body.SyncAggregate.SyncCommitteeBits.SetBitAt(i, true)
-	}
+		for i := uint64(0); i < params.BeaconConfig().MinSyncCommitteeParticipants; i++ {
+			block.Block.Body.SyncAggregate.SyncCommitteeBits.SetBitAt(i, true)
+		}
 
-	signedBlock, err := blocks.NewSignedBeaconBlock(block)
-	require.NoError(l.T, err)
+		signedBlock, err := blocks.NewSignedBeaconBlock(block)
+		require.NoError(l.T, err)
 
-	h, err := signedBlock.Header()
-	require.NoError(l.T, err)
+		h, err := signedBlock.Header()
+		require.NoError(l.T, err)
 
-	err = state.SetLatestBlockHeader(h.Header)
-	require.NoError(l.T, err)
-	stateRoot, err := state.HashTreeRoot(ctx)
-	require.NoError(l.T, err)
+		err = state.SetLatestBlockHeader(h.Header)
+		require.NoError(l.T, err)
+		stateRoot, err := state.HashTreeRoot(ctx)
+		require.NoError(l.T, err)
 
-	// get a new signed block so the root is updated with the new state root
-	block.Block.StateRoot = stateRoot[:]
-	signedBlock, err = blocks.NewSignedBeaconBlock(block)
-	require.NoError(l.T, err)
+		// get a new signed block so the root is updated with the new state root
+		block.Block.StateRoot = stateRoot[:]
+		signedBlock, err = blocks.NewSignedBeaconBlock(block)
+		require.NoError(l.T, err)
 
-	l.State = state
-	l.AttestedState = attestedState
-	l.AttestedHeader = attestedHeader
-	l.Block = signedBlock
-	l.Ctx = ctx
+		l.State = state
+		l.AttestedState = attestedState
+		l.AttestedHeader = attestedHeader
+		l.Block = signedBlock
+		l.Ctx = ctx
 
-	return l
+		return l
```

Comment:
this part is identical in both branches and can be extracted outside of the if/else